crates/sui-json-rpc/src/name_service.rs

```diff
@@ -36,8 +37,20 @@ pub struct Domain {
     labels: Vec<String>,
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
-pub struct DomainParseError;
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq, Error)]
+pub enum DomainParseError {
+    #[error("Domain is too long")]
```

Comment:
can we specify the max allowed? Domain values can contain up to XX...?

Reply:
That's probably a good idea, unless these values are being changed very often.